### DISCUSSION TITLE
Getting vite-plugin-vue-component-preview to work on Nuxt 3

### DISCUSSION THREAD

**Discussion by MahmoodKhalil57** [nuxt/nuxt#16075] [Questions] [answered]
Hello, after starting a conversation on the discord server, I was encouraged to ask over here and provide more details.
https://discord.com/channels/473401852243869706/1048310293928890488

Im trying to johnsoncodehk's component preview to work on Nuxt, but it is not working for me and Im not sure why, Im not very technical so Ill provide as much detail as I can, but I hope a kind soul would investigate this further and potentially produce a proper bug ticket for this if needed.
https://github.com/johnsoncodehk/vite-plugin-vue-component-preview

If you want to jump straight in, I provided two examples, first (vanilla vue) working as expected, and the second (Nuxt) not working:

An example of the plugin working in a vanilla vue project by antfu (NOT nuxt) can be found here: 

git clone https://github.com/antfu/vitesse.git -> npm install -> open the project in VsCode -> download "Vue Language Features (Volar)" extension -> open any vue file (component or page) -> in the explorer tab click on "vue component preview" (screenshot provided) -> you can then visit http://localhost:3333/__preview/src/components/TheFooter.vue to verify it is working in browser too
![image](https://user-images.githubusercontent.com/46604753/205468193-ee10ef84-eb3a-4a01-954b-eaab7ce3e07d.png)

An example of the plugin NOT working in a clean Nuxt project:

git clone https://github.com/MahmoodKhalil57/nuxtTest.git -> npm install -> open the project in VsCode -> download "Vue Language Features (Volar)" extension -> open any vue file (component or page) -> in the explorer tab, the "vue component preview" tab doesnt exist -> also, you can click on the nuxt icon top right and click on "preview nuxt app in external browser" (screenshot provided) -> you can then visit http://localhost:3333/__preview/src/components/testBtn.vue to verify it is not working in browser too
![image](https://user-images.githubusercontent.com/46604753/205468520-8273ad23-7df0-4ab7-b0f9-487a19f5cf67.png)

================================================================
This project ( https://github.com/MahmoodKhalil57/nuxtTest.git ) is super clean, you can go though the commit history easily, please do let me know if I did something wrong (to see the commit that I assumed should have added the plugin to nuxt yet it doesnt work: https://github.com/MahmoodKhalil57/nuxtTest/commit/64948fe5c2d26c583300cafbc5c35c377984555d)  (to see all commits since fresh nuxt install: https://github.com/MahmoodKhalil57/nuxtTest/commits/master )

**Comment by johnsoncodehk** [comment, marked as answer]
I think the way vite-plugin-vue-component-preview currently works doesn't work in SSR.

~~You can bypass Volar to test it.~~ I see that you already mentioned this.

1. `npx nuxi dev --port 3333`
2. Open http://localhost:3333/__preview/src/components/testBtn.vue in Browser
3. Edit testBtn.vue and you can see HMR not working

The way Volar supports component preview is just to open `http://localhost:{preview_port}/__preview/{component_path}` in the preview view, so it is not necessary to use vite-plugin-vue-component-preview, maybe we need to create an new nuxt plugin that can work with SSR.

**Reply by MahmoodKhalil57** [reply]
That is very unfortunate, but thank you for the reply

**Comment by johnsoncodehk** [comment]
The good news is, it's possible with Nuxt 3.2. :)
https://user-images.githubusercontent.com/16279759/217991858-726ac5ed-23fe-4e9c-a8f5-99da212b693a.mov

But a missing piece is `server.watcher.emit` cannot trigger `handleHotUpdate` with Nuxt 3.2.0. (https://github.com/johnsoncodehk/vite-plugin-vue-component-preview/blob/5df35b08c09148aeafef31c87dbb18b4cd16346f/index.ts#L30)
So HMR not working for unsaved editing.

And Volar also needs to update, please wait for the next minor release (Volar 1.1 / 1.2).
